src/CarbonAware/src/Extensions/EmissionsDataExtensions.cs

```diff
@@ -0,0 +1,85 @@
+namespace CarbonAware.Extensions;
+
+public static class EmissionsDataExtensions
+{
+    public static IEnumerable<EmissionsData> RollingAverage(this IEnumerable<EmissionsData> data, TimeSpan windowSize = default, TimeSpan tickSize = default)
+    {
+        if(data.Count() == 0){ yield break; }
+
+        if(windowSize == default)
+        {
+           foreach(var d in data){ yield return d; }
+           yield break;
+        }
+
+        var q = new Queue<EmissionsData>();
+        var _data = data.GetEnumerator();
+        _data.MoveNext();
+        EmissionsData current = _data.Current;
+        #nullable enable
```

Comment:
If the whole library isn't nullable, then by default, objects like EmissionsData are nullable and the question mark isn't necessary, especially if we have to use processor directives to turn nullable on to use the question mark.